File: cyanBat/src/main/java/at/smiech/cyanbat/util/Vector2D.java
package at.smiech.cyanbat.util;

/**
 * Classic (movement) vector class in two dimensions
 */
public class Vector2D {
	public float x;
	public float y;

	public Vector2D() {
		x = 0f;
		y = 0f;
	}

	public Vector2D(float x, float y) {
		this.x = x;
		this.y = y;
	}

	public void add(Vector2D val) {
		if(val == null) {
			throw new IllegalArgumentException("Passed 2d vector was null!");
		}
		x += val.x;
		y += val.y;
	}

	public void subtract(Vector2D val) {
		if(val == null) {
			throw new IllegalArgumentException("Passed 2d vector was null!");
		}
		x -= val.x;
		y -= val.y;
	}
}

File: cyanBat/src/main/java/at/smiech/cyanbat/util/Vector2D.kt
package at.smiech.cyanbat.util

/**
 * Classic (movement) vector class in two dimensions
 */
data class Vector2D(
    var x: Float = 0f,
    var y: Float = 0f
) {
    operator fun plus(other: Vector2D): Vector2D = Vector2D(x + other.x, y + other.y)
}
